src/gromacs/nblib/topology.cpp

```diff
@@ -182,8 +298,41 @@ gmx::ListOfLists<int> TopologyBuilder::createExclusionsListOfLists() const
     return exclusionsListOfListsGlobal;
 }
 
+Topology::InteractionData TopologyBuilder::createInteractionData(const detail::ParticleSequencer& particleSequencer)
+{
+    auto create = [](auto& output, const auto& molecules, const auto& sequencer) {
+        using BondType = typename std::decay_t<decltype(output)>::type;
+        auto uniqueData = detail::eliminateDuplicateBonds(detail::aggregateBonds<BondType>(molecules));
+        auto& uniqueIndices       = std::get<0>(uniqueData);
+        auto& uniqueBondInstances = std::get<1>(uniqueData);
+
+        // add data about BondType instances
+        output.bondInstances = uniqueBondInstances;
+
+        // add data about interaction pair indices
+        auto& indexTriples = output.indices;
+        indexTriples.resize(uniqueIndices.size());
+        auto pairIndices = detail::sequencePairIDs<BondType>(molecules, sequencer);
+        std::transform(begin(pairIndices), end(pairIndices), begin(uniqueIndices),
+                       begin(indexTriples), [](auto pairIndex, auto bondIndex) {
+                           return std::make_tuple(std::get<0>(pairIndex), std::get<1>(pairIndex), bondIndex);
+                       });
+    };
+
+    Topology::InteractionData interactionData;
+
+    // inject molecule_ and particleSequencer arguments
+    auto create_this = [this, &particleSequencer, f = create](auto& output) {
+        f(output, this->molecules_, particleSequencer);
```

Comment:
I don't see the utility of this indirection. Why separate `create` as a separate lambda rather than embedding here?

Reply:
I have some defensive reflexes against lambda reference capture, so I separated into a lambda with and without ref capture. I changed it to a single lambda and the improved readability is more than worth it.